dask/array/creation.py

```diff
@@ -452,14 +578,8 @@ def eye(N, chunks, M=None, k=0, dtype=float):
     if M % chunks != 0:
         hchunks.append(M % chunks)
 
-    for i, vchunk in enumerate(vchunks):
-        for j, hchunk in enumerate(hchunks):
-            if (j - i - 1) * chunks <= k <= (j - i + 1) * chunks:
-                eye[name_eye, i, j] = (np.eye, vchunk, hchunk, k - (j - i) * chunks, dtype)
-            else:
-                eye[name_eye, i, j] = (np.zeros, (vchunk, hchunk), dtype)
-    return Array(eye, name_eye, shape=(N, M),
-                 chunks=(chunks, chunks), dtype=dtype)
```

Comment:
I actually find the previous implementation to be a bit easier to understand and review in the future.  It's more lines of code, yes, but it doesn't require the reader to understand the `from_block_function` function.  These for loops are, to me at least, not a barrier to understanding.  I'm inclined to leave this as is.